spec.md

```diff
@@ -147,12 +147,20 @@ that contains both context and data).
   * MUST be a non-empty string
 
 ### source
-* Type: Object
-* Description: This describes the software instance that emits the event at
-  runtime (i.e. the producer). It contains sub-properties (listed below)
+* Type: String
+* Description: This property identifies the source of the event as defined
+  by the publishing organization. The value SHOULD be a URI. The source property
+  value MUST uniquely identify the source within the scope set by the "namespace"
+  property. While the value MAY be a interpreted as a URL, the purpose of
+  it to logically identify the source and not to establish the location of the 
+  event source on the network.
```

Comment:
This certainly clarifies the description more; however, it raises as many questions as it answers.  How do we expect that the source will be used?  We want to consider use cases for tools, providers, and enabling an action to call back to the source, when appropriate.

Given these open questions, it is hard to evaluate whether the option suggested will be effective.

Reply:
The source identifies who emits the event, scoped by the namespace. The source can be any string that makes sense for the publishing organization and their subscribers (in the extreme it could hold a JSON object), but it's suggested that it SHOULD be a URI. I will lay out in the call that the source property information by itself cannot be sufficient to determine a callback path to the source in non-trivial scenarios. The statement "MAY be a interpreted as a URL" explicitly enables interpretation of the identifier as a callback location if an implementation wants that.